sas9.4-viya/nestedtemplates/vm_linux_extension.json

```diff
@@ -0,0 +1,65 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "location": {
+      "metadata": {
+        "description": "The Location For the resources"
+      },
+      "type": "string"
+    },
+    "vmName": {
+      "metadata": {
+        "description": "The name of the vm"
+      },
+      "type": "string"
+    },
+    "commandToExecute": {
+      "type": "string"
+    },
+    "_artifactsLocation": {
+      "type": "string",
+      "metadata": {
+        "description": "The base URI where artifacts required by this template are located. When the template is deployed using the accompanying scripts, a private location in the subscription will be used and this value will be automatically generated."
+      }
+    },
+    "_artifactsLocationSasToken": {
+      "type": "securestring",
+      "metadata": {
+        "description": "The sasToken required to access _artifactsLocation."
+      }
+    }
+  },
+  "variables": {
+    "extensionName": "[concat(parameters('vmName'),'InitStartup')]"
+  },
+  "resources": [
+    {
+      "type": "Microsoft.Compute/virtualMachines/extensions",
+      "apiVersion": "2019-07-01",
+      "name": "[concat(parameters('vmName'), '/', variables('extensionName'))]",
+      "location": "[parameters('location')]",
+      "properties": {
+        "publisher": "Microsoft.Azure.Extensions",
+        "type": "CustomScript",
+        "typeHandlerVersion": "2.0",
+        "autoUpgradeMinorVersion": true,
+        "settings": {
+          "skipDos2Unix": false
+        },
+        "protectedSettings": {
+          "commandToExecute": "[parameters('commandToExecute')]",
+          "fileUris": [
+            "[concat(parameters('_artifactsLocation'),parameters('_artifactsLocationSasToken'))]"

```

Comment:
two things:
1) use the uri() function (remove concat)
2) follow the pattern used for artifacts in the main template, e.g.
uri(parameters(_artifactsLocation), concat('artifact', parameters(_artifactsLocationSasToken)))

Reply:
here, as we are using the nested templated for role assignment for the virtual machines, we have constructed the value for each extension in the main template variables itself and passed the value to the extension nested template.

Reply:
done